src/CoreWCF.Primitives/src/CoreWCF/Security/NegotiationTokenAuthenticatorState.cs

```diff
@@ -10,11 +11,9 @@ internal class NegotiationTokenAuthenticatorState : IDisposable
     {
         private bool _isNegotiationCompleted;
         private SecurityContextSecurityToken _serviceToken;
-        private readonly object _thisLock;
+        private readonly AsyncLock _asyncLock = new AsyncLock();
 
-        public NegotiationTokenAuthenticatorState() => _thisLock = new object();
-
-        public object ThisLock => _thisLock;
+        public AsyncLock AsyncLock => _asyncLock;
 
```

Comment:
You can actually do this:
```c#
public AsyncLock AsyncLock { get; } = new AsyncLock();
```
It saves having an explicit backing field.

Reply:
I removed the redundant backing field declaration